compiler_opt/rl/compilation_runner.py

```diff
@@ -102,13 +104,22 @@ class WorkerCancellationManager:
   managing resources.
   """
 
-  def __init__(self):
+  @dataclasses.dataclass
+  class ProcData:
```

Comment:
a thought: the motivating scenario here is the validator. For validation, we're actually OK to let compilation run longer than x seconds - because the goal is to get a thorough idea of what-if this model were shipped. So, how about:
- no ProcData
- just have the validator use a very large timeout, like 20 minutes (i.e. ~half of that in real compilation time)